adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -461,13 +460,11 @@ def publish(self, topic, msg, retain=False, qos=0):
             0 <= qos <= 1
         ), "Quality of Service Level 2 is unsupported by this library."
 
-        pub_hdr_fixed = bytearray()  # fixed header
-        pub_hdr_fixed.extend(MQTT_PUB)
-        pub_hdr_fixed[0] |= retain | qos << 1  # [3.3.1.2], [3.3.1.3]
+        # fixed header. [3.3.1.2], [3.3.1.3]
+        pub_hdr_fixed = bytearray([MQTT_PUB[0] | retain | qos << 1])
 
-        pub_hdr_var = bytearray()  # variable header
-        pub_hdr_var.append(len(topic) >> 8)  # Topic length, MSB
-        pub_hdr_var.append(len(topic) & 0xFF)  # Topic length, LSB
+        # variable header = 2-byte Topic length (big endian)
+        pub_hdr_var = struct.pack(">H", len(topic))
```

Comment:
Running the example `simpletest` on `Adafruit CircuitPython 5.3.1 on 2020-07-13; Adafruit PyPortal with samd51j20` throws the following error in the REPL:

```
Subscribed to test/topic with QOS level 0
Publishing to test/topic
Traceback (most recent call last):
  File "code.py", line 116, in <module>
  File "/lib/adafruit_minimqtt/adafruit_minimqtt.py", line 468, in publish
AttributeError: 'bytes' object has no attribute 'extend'
```

Instead of `extend` on L468, we may want to `append`

Reply:
I feel like this is a borderline rookie mistake. I tried `append()` instead, but I still got the same error `AttributeError: 'bytes' object has no attribute 'append'` However `pub_hdr_var += topic.encode()` does what we want. The real problem comes from the change to using `struct.pack()` which returns an immutable `bytes` object, where methods like `append()` or `extend(bytes([0x00]))` (notice the need for conversion from `int` to `bytes` when using `extend()`) can only apply to a mutable `bytearray` object. So, I've wrapped the `struct.pack()` into a `bytearray(struct.pack())` which allows us to keep using the many `pub_hdr_var.append()` calls after L468.